src/SprykerEco/Yves/Payone/Checkout/Process/StepFactory.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Yves\Payone\Checkout\Process;
+
+use Spryker\Yves\Checkout\DataContainer\DataContainer;
+use Spryker\Yves\Checkout\Dependency\Client\CheckoutToQuoteBridge;
+use Spryker\Yves\Kernel\AbstractFactory;
+use Spryker\Yves\ProductBundle\Grouper\ProductBundleGrouper;
+use Spryker\Yves\StepEngine\Process\StepCollection;
+use Spryker\Yves\StepEngine\Process\StepCollectionInterface;
+use Spryker\Yves\StepEngine\Process\StepEngine;
+use SprykerEco\Yves\Payone\Checkout\Process\Steps\InitExpressCheckoutStep;
+use SprykerEco\Yves\Payone\Checkout\Process\Steps\LoadExpressCheckoutDetailsStep;
+use SprykerEco\Yves\Payone\Checkout\Process\Steps\PlaceOrderStep;
+use SprykerEco\Yves\Payone\Checkout\Process\Steps\SuccessStep;
+use SprykerEco\Yves\Payone\Checkout\Process\Steps\SummaryStep;
+use SprykerEco\Yves\Payone\Handler\ExpressCheckout\QuoteHydrator;
+use SprykerEco\Yves\Payone\PayoneDependencyProvider;
+use SprykerEco\Yves\Payone\Plugin\Provider\PayoneControllerProvider;
+
+class StepFactory extends AbstractFactory
+{
+
+    /**
+     * @param \Spryker\Yves\StepEngine\Process\StepCollectionInterface $stepCollection
+     *
+     * @return \Spryker\Yves\StepEngine\Process\StepEngine
+     */
+    public function createStepEngine(StepCollectionInterface $stepCollection)
```

Comment:
why you need this argument?
I would use directly method `createStepCollection`.
It still allows you to modify functionality on project level